src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -287,6 +322,31 @@ public final class EnhancedQueueExecutor extends EnhancedQueueExecutorBase6 impl
     private static final long activeCountOffset;
     private static final long peakQueueSizeOffset;
 
+    // GraalVM should initialize this class at run time
+    private static final class RuntimeFields {
+        private static final int unsharedObjectsSize;
+        private static final int unsharedLongsSize;
+
+        private static final long headOffset;
+        private static final long tailOffset;
+        private static final long threadStatusOffset;
+
+        static {
+            int cacheLine = CacheInfo.getSmallestDataCacheLineSize();
+            if (cacheLine == 0) {
+                // guess
+                cacheLine = 64;
+            }
+            int longScale = unsafe.arrayIndexScale(long[].class);
+            int objScale = unsafe.arrayIndexScale(Object[].class);
+            unsharedObjectsSize = cacheLine / objScale * (numUnsharedObjects + 1);
+            unsharedLongsSize = cacheLine / longScale * (numUnsharedLongs + 1);
+            headOffset = unsafe.arrayBaseOffset(Object[].class) + (cacheLine / objScale);
```

Comment:
Assuming 64 bytes cache line, COOPS and 0 array base offset, it means 

headOffset := 0 + (64 /4 ) = 16
tailOffset := 0 + (64 /4 ) * 2 = 32

which means that if both are within `unsharedObjects`, their distance is just 16 bytes, which is not enough to prevent false-sharing between them

They should be in 2 separate cache lines, if we don't want to interact with each others, but if we expect, in the common scenario that they would be invalidated and used always together, it's ok.

Reply:
You're right, I don't know why I did this division. I guess I was thinking of array indexes rather than bytes for some reason.